extensions/amp-render/1.0/amp-render.js

```diff
@@ -335,6 +336,37 @@ export class AmpRender extends BaseElement {
       .then((html) => dict({'__html': html}));
   }
 
+  /**
+   * Returns the non-empty node count in a template (defined as a `template`
+   * or `script` element). This is required to establish if the template content has
+   * a single wrapper element and if so we need to include it while rendering the
+   * template. For more info, see https://github.com/ampproject/amphtml/issues/35401.
+   * TODO(dmanek): Observe rewrapping at a lower level in BaseTemplate.
+   * @param {?Element} template
+   * @return {number} count of non-empty child nodes
+   * @private
+   */
+  getTemplateNonEmptyNodeCount_(template) {
```

Comment:
nit: can we hoist this as a standalone function that takes a `document` and a `template`?